src/txmempool.cpp

```diff
@@ -637,9 +637,12 @@ void CTxMemPool::removeForBlock(const std::vector<CTransactionRef>& vtx, unsigne
         if (it2 != mapWithdrawsSpentToTxid.end()) {
             txiter txit = mapTx.find(it2->second);
             assert(txit != mapTx.end());
+            const CTransaction& tx = txit->GetTx();
             setEntries stage;
             stage.insert(txit);
             RemoveStaged(stage, true);
+            removeRecursive(tx, MemPoolRemovalReason::BLOCK);
```

Comment:
I'm probably not familiar enough with the code to properly review this. If the transaction is in the block and in the mempool then then it is removed from the mempool with `removeRecursive` and not with `RemoveStaged` which is different to the behavior for non-pegin transactions. Maybe it's better to only call `removeRecursive` for peg-ins that are in the mempool but not in the block.

Reply:
Not sure what you're saying here. The code here looks for conflicting peg-ins, removes those(since they aren't removed earlier, the earlier code is only concerned about elements outputs), then directly travels its descendants to remove them. The other block of code uses `removeConflicts` which won't identify the mapping between the conflicting peg-in transactions.

Reply:
Yeah, I'm talking about something else, but I think I understand it now.

My point was that `removeRecursive` (and `ClearPrioritisation`) is now also applied to confirming transactions because tx can be a conflicting tx or the tx that's confirming. Whereas regularly it would only apply to unconfirmed conflicting transactitons (due to https://github.com/ElementsProject/elements/pull/297/files#diff-ca74c4b28865382863b8fe7633a85cd6R596).

However, reading into RemoveStaged it appears like the confirming tx is already removed from `mapWithdrawsSpentToTxid` and what we're iterating over in the codeblock are just conflicting unconfirmed peg-ins.

Reply:
Correct.